core/slim_globals.h

```diff
@@ -445,8 +445,7 @@ extern EidosValue_String_SP gStaticEidosValue_StringT;
 	
 #define SLIM_TSK_INDIVIDUAL_ALIVE       ((uint32_t)(1 << 16))
 #define SLIM_TSK_INDIVIDUAL_REMEMBERED  ((uint32_t)(1 << 17))
-// DEPRECATED:
-// #define SLIM_TSK_INDIVIDUAL_FIRST_GEN   ((uint32_t)(1 << 18))
```

Comment:
@petrelharp any reason to keep the old deprecated flag?  Is there any potential issue with @hyanwong re-using the same (1 << 18) value for his new flag?

Reply:
Thanks for picking that up. I was going to ask...

Reply:
No, I don't think so. Maybe pyslim needs to know what it is to be able to read old trees files, but we don't need it here.

Reply:
Besides, the flags have similar meanings: "this individual is in the tree sequence just 'cause it happens to still be here", sort of.

Reply:
Hmm.  I guess pyslim can have logic that says "if this flag is set and the tree sequence version is less than X, then it means "first gen", otherwise it means "retained".  SLiM would need that logic too, perhaps, in its tree sequence reading code?  Do we do anything with these flags when we read a tree sequence in?  Should we?

Reply:
This last question remains unanswered.  @petrelharp, do we need to do anything with this new flag (or with the possible existence of the deprecated SLIM_TSK_INDIVIDUAL_FIRST_GEN flag in an old file, now that we are re-using the same flag value) when we read in a tree sequence?  It makes me nervous re-using the same value.  If an old tree sequence has individuals marked with SLIM_TSK_INDIVIDUAL_FIRST_GEN, would they then be marked as "retained" when we read that old tree sequence in the new SLiM?  And if so, is that OK, or should we worry about it?  There are lots of people still using old versions of SLiM, so this is a real concern.  (Somebody recently published a paper that cited SLiM 1.8!)

Reply:
Well, let's see:
- SLiM does not use the `SLIM_TSK_INDIVIDUAL_FIRST_GEN` flag for anything; we know this because it's actually commented out of the source code.
- pyslim [does not use it for anything either](https://github.com/tskit-dev/pyslim/search?q=FIRST_GEN), although [this method](https://github.com/tskit-dev/pyslim/blob/9448c91ede9bf40fb40ab0dc621b196d3cc1bf47/pyslim/slim_tree_sequence.py#L761) exists to help people find the FIRST_GEN individuals.

So: tree sequences produced by old versions of SLiM will have the FIRST_GEN individuals in the tree sequence. These are read into SLiM no problem. And, these are read into pyslim, and retain the FIRST_GEN flags, but the only thing this affects is how people *use* the tree sequence: sometimes they'd get confused about why there's individuals that aren't alive at the end of the simulation. It doesn't affect how anything works, or even any recommended workflows.

If we re-use the flag, but rename it RETAINED, then what will happen if we use an old tree sequence with the FIRST_GEN flag? Well, in SLiM  those FIRST_GEN individuals will act as if they were RETAIN'ed. Which is fine, since they actually were (just for a different reason). And again, in pyslim, we don't actually use the REMEMBERED flag for anything, so this won't mess anything up.

So - I think this is fine - we can re-use the flag, and given that the two flags actually mean more or less the same thing (that the individual doesn't necessarily have nodes that are samples), we *should* re-use it.

Reply:
Hurrah. Merge away!

Reply:
OK.  Thanks for the reassurance; this makes sense.  I'll merge now.